frontend/src/metabase/auth/components/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { t } from "ttag";
+import { Form, Formik } from "formik";
+import * as Yup from "yup";
+import useForm from "metabase/core/hooks/use-form";
+import FormCheckBox from "metabase/core/components/FormCheckBox";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import FormField from "metabase/core/components/FormField";
+import FormInput from "metabase/core/components/FormInput";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import { LoginData } from "../../types";
+
+const LdapSchema = Yup.object().shape({
+  username: Yup.string().required(t`required`),
+  password: Yup.string().required(t`required`),
+  remember: Yup.boolean(),
+});
+
+const PasswordSchema = LdapSchema.shape({
+  username: Yup.string()
+    .required(t`required`)
+    .email(t`must be a valid email address`),
+});
+
+export interface LoginFormProps {
+  isLdapEnabled: boolean;
+  hasSessionCookies: boolean;
+  onSubmit: (data: LoginData) => void;
+}
+
+const LoginForm = ({
+  isLdapEnabled,
+  hasSessionCookies,
+  onSubmit,
+}: LoginFormProps): JSX.Element => {
+  const initialValues: LoginData = {
+    username: "",
+    password: "",
+    remember: !hasSessionCookies,
+  };
+  const handleSubmit = useForm(onSubmit);
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={isLdapEnabled ? LdapSchema : PasswordSchema}
+      isInitialValid={false}
+      onSubmit={handleSubmit}
+    >
+      <Form>
+        <FormField
+          name="username"
+          title={
+            isLdapEnabled ? t`Username or email address` : t`Email address`
+          }
+        >
+          <FormInput
+            name="username"
+            type={isLdapEnabled ? "input" : "email"}
+            placeholder="nicetoseeyou@email.com"
+            autoFocus
+            fullWidth
+          />
+        </FormField>
+        <FormField name="password" title={t`Password`}>
```

Comment:
I'm wondering wdyt about making `FormField` something like `FormField` wrapper which will be reused inside `FormInput`, `FormCheckbox`, etc?

So that when you create forms you don't need to wrap its items with FormField and sync its names which is a potential area to make a mistake

Reply:
I think that makes sense but would make customization much hander. I'd rather call this v1 API, see how it goes, and then consolidate components if this would be not needed.